### PR TITLE
added ShoppingCart and Ticket

### PR DESCRIPTION
**Pull request by VitaliiLeshchenko**

### CODE REVIEW

File: src/main/java/cinema/dao/impl/ShoppingCartDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+            transaction = session.beginTransaction();
+            session.save(shoppingCart);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't save shoppingCart : " + shoppingCart, e);
+        }
+        return shoppingCart;
```
Comment (dmytro-hryhoruk) [inline src/main/java/cinema/dao/impl/ShoppingCartDaoImpl.java:30]: agreed not to use try-with resources in insert operaitons

File: src/main/java/cinema/dao/impl/ShoppingCartDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+                    "FROM ShoppingCart sc "
+                            + "JOIN FETCH sc.user "
+                            + "where sc.user.id = :userId", ShoppingCart.class);
+            query.setParameter("userId", user.getId());
+            ShoppingCart shoppingCart = query.getSingleResult();
+            for (Ticket ticket : shoppingCart.getTickets()) {
+                Hibernate.initialize(ticket.getMovieSession());
+                Hibernate.initialize(ticket.getMovieSession().getMovie());
+                Hibernate.initialize(ticket.getMovieSession().getCinemaHall());
+            }
```
Comment (dmytro-hryhoruk) [inline src/main/java/cinema/dao/impl/ShoppingCartDaoImpl.java:46]: each initialize it's a separate select operation, can we avoid that?

File: src/main/java/cinema/dao/impl/ShoppingCartDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+            return shoppingCart;
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't get shoppingCart by User", e);
+        }
+    }
+
+    @Override
+    public void update(ShoppingCart shoppingCart) {
+        Transaction transaction = null;
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
```
Comment (dmytro-hryhoruk) [inline src/main/java/cinema/dao/impl/ShoppingCartDaoImpl.java:56]: same about try-with resources

File: src/main/java/cinema/dao/impl/TicketDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+import cinema.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+
+@Dao
+public class TicketDaoImpl implements TicketDao {
+    @Override
+    public Ticket add(Ticket ticket) {
+        Transaction transaction = null;
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
```
Comment (dmytro-hryhoruk) [inline src/main/java/cinema/dao/impl/TicketDaoImpl.java:16]: and here

File: src/main/java/cinema/model/ShoppingCart.java [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+import javax.persistence.Id;
+import javax.persistence.ManyToMany;
+import javax.persistence.MapsId;
+import javax.persistence.OneToOne;
+
+@Entity
+public class ShoppingCart {
+    @Id
+    private Long id;
+    @ManyToMany(cascade = CascadeType.ALL, fetch = FetchType.LAZY)
```
Comment (dmytro-hryhoruk) [inline src/main/java/cinema/model/ShoppingCart.java:17]:
```suggestion
    @OneToMany(cascade = CascadeType.ALL, fetch = FetchType.LAZY)
```

Comment (dmytro-hryhoruk) [reply src/main/java/cinema/model/ShoppingCart.java:17]: we can't have same ticket in two different shopping carts

Comment (VitaliiLeshchenko) [reply src/main/java/cinema/model/ShoppingCart.java:17]: done

File: src/main/java/cinema/dao/impl/ShoppingCartDaoImpl.java (hunk 1 above)
Comment (VitaliiLeshchenko) [reply src/main/java/cinema/dao/impl/ShoppingCartDaoImpl.java]: changed

File: src/main/java/cinema/dao/impl/ShoppingCartDaoImpl.java (hunk 2 above)
Comment (VitaliiLeshchenko) [reply src/main/java/cinema/dao/impl/ShoppingCartDaoImpl.java]: changed

File: src/main/java/cinema/dao/impl/TicketDaoImpl.java (hunk 4 above)
Comment (VitaliiLeshchenko) [reply src/main/java/cinema/dao/impl/TicketDaoImpl.java]: changed

File: src/main/java/cinema/model/ShoppingCart.java (hunk 5 above)
Comment (VitaliiLeshchenko) [reply src/main/java/cinema/model/ShoppingCart.java]: changed

File: src/main/java/cinema/dao/impl/ShoppingCartDaoImpl.java (hunk 3 above)
Comment (VitaliiLeshchenko) [reply src/main/java/cinema/dao/impl/ShoppingCartDaoImpl.java:55]: changed

**Pull request merged**